docker-compose.yml

```
@@ -0,0 +1,45 @@
+version: '3'
+
+services:
+  production:
+    build:
+      context: .
+      dockerfile: Dockerfile
+      target: production
+    image: todo_app:prod
+    ports:
+      - "5800:8000"
+    env_file:
+      - ".env"
+    profiles:
+      - prod
+
+  development:
+    build: 
+      context: .
+      dockerfile: Dockerfile
+      target: development
+    image: todo_app:dev
+    ports:
+      - "5800:5000"
+    env_file:
+      - ".env"
+    volumes:
+      - type: bind
+        source: ./todo_app
+        target: /app/todo_app
+    profiles:
+      - dev
+  
+  testing:
+    build:
+      context: .
+      dockerfile: Dockerfile
+      target: testing
+    image: todo_app:test
+    volumes:
+      - type: bind
+        source: ./todo_app
+        target: /app/todo_app
```

Personal preference whether you prefer the concise or the clearer version, but this is equivalent to just:
```suggestion
      - ./todo_app:/app/todo_app
```
The fact that the left-hand-side is a path is enough to signal a bind mount. I know some people like this shorthand, and others find it unclear and unhelpful!

---

I think as I'll likely use this repo as a reference for other projects in the future I'll leave it where everything is explicitly spelled out for clarity. Good to know though!